### ISSUE TITLE
BUG?: `rebin` requires dwell time for EELS signals

### ISSUE THREAD

**Issue by woozey** [hyperspy/hyperspy#1957]
I got my EELS map loaded from FEI's .emi file. Trying to bin it down got following:
``` python
>>> eels_before.rebin(scale=(1., 1., 8.))

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-119-395d823fc946> in <module>()
----> 1 eels_before.rebin(scale=(1., 1., 8.))

~/anaconda/lib/python3.5/site-packages/hyperspy/_signals/eels.py in rebin(self, new_shape, scale, crop, out)
   1349         time_factor = np.prod([factors[axis.index_in_array]
   1350                                for axis in m.axes_manager.navigation_axes])
-> 1351         mdeels = m.metadata.Acquisition_instrument.TEM.Detector.EELS
   1352         m.get_dimensions_from_data()
   1353         if "Acquisition_instrument.TEM.Detector.EELS.dwell_time" in m.metadata:

~/anaconda/lib/python3.5/site-packages/hyperspy/misc/utils.py in __getattribute__(self, name)
    336             name = name.decode()
    337         name = slugify(name, valid_variable_name=True)
--> 338         item = super(DictionaryTreeBrowser, self).__getattribute__(name)
    339         if isinstance(item, dict) and '_dtb_value_' in item and "key" in item:
    340             return item['_dtb_value_']

AttributeError: 'DictionaryTreeBrowser' object has no attribute 'EELS'
```
It seems from the code that the functions trying to access "Acquisition_instrument.TEM.Detector.EELS.dwell_time" in metadata, which is not there for my signal:
``` python
>>> eels_before.metadata

├── Acquisition_instrument
│   └── TEM
│       ├── Detector
│       │   └── Camera
│       │       └── exposure = 0.000004
│       ├── Stage
│       │   ├── tilt_alpha = 0.0
│       │   ├── tilt_beta = 0.0
│       │   ├── x = -0.163507
│       │   ├── y = -0.01909
│       │   └── z = 0.081091
│       ├── acquisition_mode = STEM
│       ├── beam_energy = 300.0
│       ├── camera_length = 77.0
│       ├── magnification = 115000.0
│       └── microscope = Titan 300 kV D3096 SuperTwin
├── General
│   ├── authors = Service
│   ├── date = 2018-02-20
│   ├── original_filename = 13.39.50 Drift corrected spectrum image_before_1.ser
│   ├── time = 13:38:33
│   └── title = 
└── Signal
    ├── binned = True
    └── signal_type = EELS
```
So I guess, if dwell time isn't there,  `rebin` should just ignore it or ask user to specify dwell time in the correct place in the metadata.

**Comment by woozey** [comment]
FYI: @k8macarthur

**Comment by k8macarthur** [comment]
Looks like you have a dwell time in your metadata. Just not in the EELS subcategory. I guess line 1351 just needs to have the .EELS removed and moved into  the if statement later.  Should be a quick fix. I'll add it into my robin improvements I'm working on at the moment.

**Comment by woozey** [comment]
Yep, for now I have fixed it just adding the required node.

**Comment by francisco-dlp** [comment]
I also think that this is a bug: ``rebin`` should work even if dwell time is not defined. However, in that case it should add a warning to the log.

Also, it seems that the metadata is incorrectly mapped. @woozey, could you share the ``eels_before.original_metadata`` too?

**Comment by woozey** [comment]
@francisco-dlp , here it is:
```python
>>> eels_before.original_metadata

├── ObjectInfo
│   ├── AcquireDate = Tue Feb 20 13:38:33 2018
│   ├── AcquireInfo
│   │   ├── DwellTimePath = 0.000004
│   │   ├── FrameTime = 20.1327
│   │   └── Magnification = 114999 X
│   ├── DetectorRange
│   │   ├── EndX = 10.0000
│   │   ├── EndY = 10.0000
│   │   ├── StartX = -10.0000
│   │   └── StartY = -10.0000
│   ├── ExperimentalConditions
│   │   └── MicroscopeConditions
│   │       ├── AcceleratingVoltage = 300000
│   │       ├── Tilt1 = 0.000000
│   │       └── Tilt2 = 0.000000
│   ├── ExperimentalDescription
│   │   ├── C1 Aperture_um = 2000.0
│   │   ├── C2 Aperture_um = 70.0
│   │   ├── C2 lens_% = 15.66
│   │   ├── C3 Aperture_um = 1000.0
│   │   ├── C3 lens_% = 34.966
│   │   ├── Camera length_m = 0.077
│   │   ├── Defocus_um = -0.101
│   │   ├── Diffraction lens_% = 41.6
│   │   ├── Emission_uA = 321.8
│   │   ├── Extraction voltage_V = 3650.0
│   │   ├── Filter drift tube_eV = 0.0
│   │   ├── Filter ht offset_eV = 0.0
│   │   ├── Filter mode = Spectroscopy
│   │   ├── Filter prism shift_eV = -0.0
│   │   ├── Filter selected aperture = 6 mm
│   │   ├── Filter selected dispersion_eV/Channel = 0.1
│   │   ├── Filter slit state = Retracted
│   │   ├── Filter slit width_eV = 20.0
│   │   ├── Filter total energy loss_eV = 0.0
│   │   ├── Gun lens = 784
│   │   ├── Gun type = FEG
│   │   ├── High tension_kV = 300.0
│   │   ├── Illuminated Area Diameter_um = 0.0
│   │   ├── Magnification_x = 115000.0
│   │   ├── Microscope = Microscope Titan 300 kV D3096 SuperTwin
│   │   ├── Mode =  STEM nP SA Zoom Diffraction
│   │   ├── OBJ Aperture = retracted
│   │   ├── Objective lens_% = 87.911
│   │   ├── Spot size = 6
│   │   ├── Stage A_deg = 0.0
│   │   ├── Stage B_deg = 0.0
│   │   ├── Stage X_um = -163.507
│   │   ├── Stage Y_um = -19.09
│   │   ├── Stage Z_um = 81.091
│   │   ├── Stem rotation correction_deg = 23.0
│   │   ├── Stem rotation_deg = -15.0
│   │   └── User = Service
│   ├── Manufacturer = FEI (STEM)
│   ├── OriginalImageUuid = 3fb4cad6-042a-44b0-ade5-add4c773897d
│   ├── TrueImageHeaderInfo
│   └── Uuid = 79ac2fde-e73e-4bc7-9376-0701771727c7
└── ser_header_parameters
    ├── ArrayLength = array([2048, 2048, 2048, ..., 2048, 2048, 2048], dtype=uint32)
    ├── ByteOrder = 18761
    ├── CalibrationDelta = array([ 0.1,  0.1,  0.1, ...,  0.1,  0.1,  0.1])
    ├── CalibrationElement = array([0, 0, 0, ..., 0, 0, 0], dtype=uint32)
    ├── CalibrationOffset = array([-20., -20., -20., ..., -20., -20., -20.])
    ├── DataType = array([6, 6, 6, ..., 6, 6, 6], dtype=uint16)
    ├── DataTypeID = 16672
    ├── Dim-1_CalibrationDelta = 8.028469816077569e-10
    ├── Dim-1_CalibrationElement = 0
    ├── Dim-1_CalibrationOffset = -2.5983460228415137e-07
    ├── Dim-1_Description = Position
    ├── Dim-1_DescriptionLength = 8
    ├── Dim-1_DimensionSize = 700
    ├── Dim-1_Units = meters
    ├── Dim-1_UnitsLength = 6
    ├── Dim-2_CalibrationDelta = -8.0284698160775504e-10
    ├── Dim-2_CalibrationElement = 10
    ├── Dim-2_CalibrationOffset = -3.5710066609387393e-08
    ├── Dim-2_Description = Position
    ├── Dim-2_DescriptionLength = 8
    ├── Dim-2_DimensionSize = 10
    ├── Dim-2_Units = meters
    ├── Dim-2_UnitsLength = 6
    ├── NumberDimensions = 2
    ├── OffsetArrayOffset = 126
    ├── PositionX = array([ -2.59433179e-07,  -2.58630332e-07,  -2.57827485e-07, ...,
         3.00151167e-07,   3.00954014e-07,   3.01756861e-07])
    ├── PositionY = array([ -2.80830203e-08,  -2.80830203e-08,  -2.80830203e-08, ...,
        -3.53086431e-08,  -3.53086431e-08,  -3.53086431e-08])
    ├── SeriesID = 407
    ├── SeriesVersion = 544
    ├── TagTypeID = array([16706, 16706, 16706, ..., 16706, 16706, 16706], dtype=uint16)
    ├── Time = array([1519130395, 1519130395, 1519130395, ..., 1519131862, 1519131862,
       1519131862], dtype=uint32)
    ├── TotalNumberElements = 7000
    ├── Unknown = array([0, 0, 0, ..., 0, 0, 0], dtype=uint16)
    └── ValidNumberElements = 7000
```

**Comment by ericpre** [comment]
Fixed in #1965.
